Include HTTPS instructions for Git and refine SSH instructions (#292)
sb2nov/mac-setup: Git/README.md

They will get added to your `.gitconfig` file.

To push code to your GitHub repositories, we will use the recommended HTTPS method. There are also instructions for using SSH. To prevent `git` from asking for your username and password every time you push a commit you can cache your credentials by running the following command, as described in the [instructions](https://help.github.com/articles/caching-your-github-password-in-git/).

```sh
git config --global credential.helper osxkeychain
```

## Using HTTPS for GitHub (recommended)

> The instructions below are referenced from [the official documentation](https://help.github.com/en/github/using-git/which-remote-url-should-i-use#cloning-with-https-urls-recommended).

Reviewer: ```suggestion
These instructions are from [the official documentation](https://help.github.com/en/github/using-git/which-remote-url-should-i-use#cloning-with-https-urls-recommended).
```

### Clone repositories using HTTPS

> After creating a new repo on GitHub, clone it using

Reviewer 2: Should this line end with a colon to introduce the example?

Author: ```suggestion
After creating a new repo on GitHub, clone it using:
```

```sh
git clone https://github.com/<username>/<repo-name>.git
```

> If you initialized with a README

Reviewer 2: This seems like an incomplete sentence? Should there be more on this line?

Author: What I meant was :

Use this command :  \<command\> if you initialized with a README

Author: ```suggestion
- if you had initialized with a README.
```

> - if you had initialized with a README.

Reviewer 2: Isn't this minus sign at the start of the line going to turn this into a list? Is that intentional?

Reviewer 2: Yep, though now you're missing a space which really should be there. In Markdown, you can escape special characters with a "\", so how about "\- if you had initialized with a README."?

Also, you can check whether your formatting works by opening your changed code in a local editor with Markdown Preview, like Atom or VS Code.

Author: Fixed. Apparently you escape the dash with a backslash(\)

Reviewer 2: Oh, LOL: I had initially included the backslash in my comment there, but I forgot that GitHub comments themselves are Markdown, so it just acted as an escape and disappears when you view my comment. :D

Reviewer 2: E.g. this is what my comment looks like to me:

```markdown
Yep, though now you're missing a space which really should be there. In Markdown, you can escape special characters with a "\", so how about "\- if you had initialized with a README."?

Also, you can check whether your formatting works by opening your changed code in a local editor with Markdown Preview, like Atom or VS Code.
```

Author: The same thing happened to me! I had a backslash in the brackets there:
`Fixed. Apparently you escape the dash with a backslash(\)`

Author: That's why I didn't understand `with a "",` in your comment

Reviewer 2: Now it all makes sense! Sorry about that!

Author: No problem at all. It was pretty funny

Author: What's left to do now?

Reviewer 2: Nothing, IMHO! Just wait for the maintainers to merge it.

Author: Okay! Thanks for guiding me through my first PR (Well technically it isn't my first one, but it's the only one in a repo that's not dead :| )

Reviewer 2: You're welcome! Thanks for your contribution, and congratulations in advance on getting your first PR merged!

> If you did not, follow the instructions in the section below

Reviewer 2: Missing a period or colon at the end of this sentence?

Author: ```suggestion
If you did not, follow the instructions in the section below.
```

### Set up a new or existing repo with HTTPS for GitHub

If you are setting up a new repo, add at least one file and commit first. Then, configure the remote and push to GitHub by running:

```sh
git remote add origin https://github.com/<username>/<repo-name>.git
git push -u origin master
```

## SSH Config for GitHub

> The instructions below are for those who wish to use SSH and not HTTPS. These instructions are referenced from [the official documentation](https://help.github.com/articles/generating-ssh-keys).

Reviewer: ```suggestion
These instructions are for those who wish to use SSH and not HTTPS, and are from [the official documentation](https://help.github.com/articles/generating-ssh-keys).
```

### Check for existing SSH keys

> First, we need to check for existing SSH keys on your computer. We do this by running:

Reviewer: ```suggestion
First check for existing SSH keys on your computer by running:
```

  IdentityFile ~/.ssh/id_rsa
```

No matter what operating system version you run you need to run this command to complete this step:

```sh
ssh-add -K ~/.ssh/id_rsa
```

### Adding a new SSH key to your GitHub account

The last step is to let GitHub know about your SSH key so GitHub can recognize you. Run this command to copy your key to your clipboard:

```sh
pbcopy < ~/.ssh/id_rsa.pub
```

Then go to GitHub and [input your new SSH key](https://github.com/settings/ssh/new). Paste your key in the "Key" text-box and pick a name that represents the computer you're currently using.

We are now ready to use SSH with GitHub!

### Clone repositories using SSH

After creating a new repo on GitHub, clone it using

```sh
git clone git@github.com:<username>/<repo-name>.git
```

> If you initialized with a README

Reviewer 2: Incomplete sentence here too?

Author: ```suggestion
 - if you had initialized with a README.
```

> If you did not, follow the instructions in the section below

Reviewer: ```suggestion
If you did not, follow the instructions in the section below.
```
